### ISSUE TITLE
Importing Python polars sets multiprocessing start method globally, breaking unrelated usages of multiprocessing

### ISSUE THREAD

**Issue by mikepii** [pola-rs/polars#5344]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Problem

When you import `polars` in Python, it can break unrelated user code or other libraries that use Python's `multiprocessing` module because it sets the global default start method or fails if it has already been set to `'fork'`.

In particular:

1. Code will not work if it relies on the `'fork'` start method and assumes the global start method is `'fork'`.
2. Code will not work as expected (always uses `'spawn'`) if it supports both the `'fork'` and `'spawn'` start methods and uses the platform's default start method to improve performance where possible.
3. `polars` cannot be imported when the default start method has been explicitly set to `fork`, even if it was set accidentally as a side-effect of calling `multiprocessing.get_start_method()`.

### Background

A few days ago, #5237 changed polars to call `multiprocessing.set_start_method('spawn')` at import time. It also changed polars to raise an exception if the global start method has already been set to anything other than `'spawn'` (including by calling `get_start_method()` on Linux, which sets it to the default of `'fork'` as a side-effect). See also some related discussion in #3144.

Unrelated user code or other libraries may be unable to support the `'spawn'` start method, e.g., due to memory limitations, or because the execution context cannot be pickled.

The default start method on macOS is '`spawn`' because `'fork'` [should be considered unsafe on macOS](https://docs.python.org/3.8/library/multiprocessing.html#contexts-and-start-methods). The '`fork`' start method is the default on Linux because it's faster and requires less memory. User code and other library code may therefore rely on the default start method to choose whichever would be best depending on the platform.

### Proposed solution

Revert #5237, and instead change `polars` to select the `'spawn'` start method using [`multiprocessing.get_context()`](https://docs.python.org/3.8/library/multiprocessing.html#multiprocessing.get_context) wherever `multiprocessing` is used in `polars`.

### Reproducible example

The default start method is changed from `'fork'` to `'spawn'` when you import `polars` on Linux. The expected result is that the default start method is still `'fork'`.

```python
>>> import polars
>>> import multiprocessing
>>> multiprocessing.get_start_method()
'spawn'
```

You cannot import `polars` after calling `multiprocessing.get_start_method()` on Linux. The expected result is that no exception is raised.

```python
>>> import multiprocessing
>>> multiprocessing.get_start_method()
'fork'
>>> import polars
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/aicompute/env/lib/python3.8/site-packages/polars/__init__.py", line 5, in <module>
    import polars.mp_check  # noqa: F401
  File "/home/aicompute/env/lib/python3.8/site-packages/polars/mp_check.py", line 51, in <module>
    mp_check()
  File "/home/aicompute/env/lib/python3.8/site-packages/polars/mp_check.py", line 42, in mp_check
    raise ImportError(
ImportError: Polars only works with python multiprocessing method set to: `mp.set_start_method("spawn")` or `mp.set_start_method("forkserver")`before importing polars. See: https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
```

### Expected behavior

See descriptions above each example

### Installed versions

<details>

```
---Version info---
Polars: 0.14.23
Index type: UInt32
Platform: Linux-5.10.124-linuxkit-x86_64-with-glibc2.29
Python: 3.8.10 (default, Jun 22 2022, 20:18:18)
[GCC 9.4.0]
---Optional dependencies---
pyarrow: 7.0.0
pandas: 1.5.1
numpy: 1.23.4
fsspec: 2022.10.0
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: 3.6.0
```

</details>

**Comment by ritchie46** [comment]
See: #5342

**Comment by mikepii** [comment]
Thanks @ritchie46, #5342 fixes problem (3). However, (1) and (2) are still broken. The proposed solution (using `multiprocssing.get_context()` to set the start method within the scope of polars rather than changing the global start method) would fix (1) and (2).

**Comment by ritchie46** [comment]
Right.. we probably should just lay off and document the multiprocessing gotcha with polars. As long as you don't use polars in your forked processes, it is fine.

**Comment by ghuls** [comment]
> Proposed solution
> 
> Revert https://github.com/pola-rs/polars/pull/5237, and instead change polars to select the 'spawn' start method using [multiprocessing.get_context()](https://docs.python.org/3.8/library/multiprocessing.html#multiprocessing.get_context) wherever multiprocessing is used in polars.

Polars doesn't use multiprocessing itself. It spawns threads for it's threadpool. Once Polars (or any other python module that spawns threads) is imported, fork shouldn't be used anymore as it will deadlock. So `multiprocessing.get_context()` is not relevant for Polars.

**Comment by ritchie46** [comment]
Yes, polars will deadlock in that other process. But users should still be able to eun polars on the main process and use multiprocessing for any other tool. As long as they don't run polars code on these new processes, it should be fine.

**Comment by ritchie46** [comment]
closed by #5347
